pkg/setting/setting.go

```diff
@@ -940,17 +941,19 @@ func (cfg *Cfg) readLDAPConfig() {
 	cfg.LDAPAllowSignup = LDAPAllowSignup
 }
 
-func (cfg *Cfg) readAWSConfig() {
+func (cfg *Cfg) handleAWSConfig() {
 	awsPluginSec := cfg.Raw.Section("aws")
 	cfg.AWSAssumeRoleEnabled = awsPluginSec.Key("assume_role_enabled").MustBool(true)
-	allowedAuthProviders := awsPluginSec.Key("allowed_auth_providers").String()
+	allowedAuthProviders := awsPluginSec.Key("allowed_auth_providers").MustString("default,keys,credentials")
 	for _, authProvider := range strings.Split(allowedAuthProviders, ",") {
 		authProvider = strings.TrimSpace(authProvider)
 		if authProvider != "" {
 			cfg.AWSAllowedAuthProviders = append(cfg.AWSAllowedAuthProviders, authProvider)
 		}
 	}
 	cfg.AWSListMetricsPageLimit = awsPluginSec.Key("list_metrics_page_limit").MustInt(500)
+	os.Setenv("ASSUME_ROLE_ENABLED", strconv.FormatBool(cfg.AWSAssumeRoleEnabled))
+	os.Setenv("ALLOWED_AUTH_PROVIDERS", allowedAuthProviders)
```

Comment:
@sunker these do not get passed to plugins unless explicitly added by the backend plugin manager.  I replaced them with constants defined in the grafana-aws-sdk

Reply:
great!